### PR TITLE
Circos revisions for paper

### PR DESCRIPTION
**Pull request by hexylena**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

Reviewers had some good suggestions that we need to fix before publication.

### CODE REVIEW

File: tools/circos/circos.xml [hunk 1]
```diff
@@ -287,6 +314,7 @@ cp '${data.data_source}' circos/data/highlight-${hi}.txt &&
                 </conditional>
                 <param name="orientation" type="boolean" label="Orient Inwards" truevalue="in" falsevalue="out"
                     help="When yes, the plot will face inwards; i.e. lowest values will be plotted to the outside"/>
+                <param name="minsize" type="float" value="0" label="Minimum size (in chromosome units)" min="0"/>
```
Comment (bgruening) [inline tools/circos/circos.xml]: maybe move this also into the macro?

File: tools/circos/macros.xml [hunk 2]
```diff
@@ -164,9 +164,20 @@ $r, $g, $b, $a
...
-            <param name="y0" type="float" value="0" label="Inside Radius" help="Values are according to minimum and maximum values of dataset. E.g. if your dataset goes from -10 to 10, y0 should be -10"/>
-            <param name="y1" type="float" value="1" label="Outside Radius" help="Values are according to minimum and maximum values of dataset. E.g. if your dataset goes from -10 to 10, y1 should be 10"/>
-
+            <conditional name="radial_position">
+                <param name="select" type="select" label="Radial Position">
+                    <option value="plot" default="True">r0/r1 | relative to plot</option>
+                    <option value="data">relative to data</option>
+                </param>
+                <when value="plot">
+                    <param name="y0" type="float" value="0" label="Inside Radius" help="Values are relative to the plot, 0 is the center of the plot, 1 is the outside of the plot."/>
```
Comment (bgruening) [inline tools/circos/macros.xml]: add min/max?

File: tools/circos/macros.xml [hunk 3]
```diff
@@ -177,8 +188,20 @@ $r, $g, $b, $a
   <xml name="backgrounds">
     <section name="sec_bkgs" title="Backgrounds">
         <repeat name="backgrounds" title="Background">
-            <param name="y0" type="float" value="0" min="0" max="1" label="Inside Radius" help="Value must be between 1 (outer edge of track) and 0 (inner edge of the track)"/>
-            <param name="y1" type="float" value="1" min="0" max="1" label="Outside Radius" help="Value must be between 1 (outer edge of track) and 0 (inner edge of the track)"/>
+            <conditional name="radial_position">
```
Comment (bgruening) [inline tools/circos/macros.xml]: is that the same as above?

File: tools/circos/macros.xml (hunk 2 above)
Comment (hexylena) [reply tools/circos/macros.xml]: We did this before and had to revert a bunch due to the unnecessary constraint. Negative numbers are valid, and numbers greater than 1 are valid, 1 is just the 'edge' of the plot (which may not be at the edge of the page), and you're free to plot at 1.2 if you need (and I have in occasion)

File: tools/circos/macros.xml (hunk 3 above)
Comment (hexylena) [reply tools/circos/macros.xml]: yeah, can macro-ize it. I usually use duplication=3 as a guide to move into a macro, but no problem to do it here.

**Comment by hexylena** [comment]
Thanks for the comments!

File: tools/circos/macros.xml [hunk 4]
```diff
@@ -290,6 +381,31 @@ $r, $g, $b, $a
     </section>
   </xml>
 
+  <xml name="link_color_action_options"
+      token_section="ERROR">
+
+    <conditional name="dynamic">
+        <param name="dynamic_select" type="select" label="Static / Dynamic Change">
+            <option value="static">Static: Change is always applied</option>
+            <option value="source">Color based on source chromosome color</option>
```
Comment (bgruening) [inline tools/circos/macros.xml:390]: dynamic is missing here in the select

File: tools/circos/stack-histogram.xml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	<command detect_errors="exit_code"><![CDATA[
+python
+	'$__tool_directory__/stack-histogram.py'
+	#for i in $input
+		'$i'
+	#end for
+	> '$output'
+	]]></command>
+	<inputs>
+		<param name="input" type="data" format="bigwig" label="BigWig files" multiple="True" help="These MUST have identical chromosomes and intervals, within their datasets. This tool is incredibly, intentionally naïve, and will crash if there are missing ranges in some bigwig files. This feels like a reasonable assumption to make (simplifies processing) at the expense of theoretical correctness, given that when you are stacking histograms they're probably coming from the same upstream pipeline with the same settings (i.e. step/width/etc.)"/>
```
Comment (bgruening) [inline tools/circos/stack-histogram.xml:18]: something is wrong here with the indentation ..

File: tools/circos/macros.xml (hunk 4 above)
Comment (hexylena) [reply tools/circos/macros.xml:390]: The second two are considered as 'dynamic', but, the repeated text label looked bad. I think I'll just remove `static:`

Comment (hexylena) [reply tools/circos/macros.xml:390]: Oh, I see now.

Comment (hexylena) [reply tools/circos/macros.xml:390]: Shouldn't that be an error in planemo? there's a conditional option, which has no select?

File: tools/circos/stack-histogram.xml (hunk 5 above)
Comment (hexylena) [reply tools/circos/stack-histogram.xml:18]: can you explain? it looks ok to me.

Comment (bgruening) [reply tools/circos/stack-histogram.xml:18]: Its using tabs not spaces imho.

File: tools/circos/macros.xml (hunk 4 above)
Comment (bgruening) [reply tools/circos/macros.xml:393]: Yes, please fil an enhancement request :)

**Comment by hexylena** [comment]
@bgruening any idea about the failures here?

**Comment by hexylena** [comment]
ahh awesome, thanks @bgruening, never would have figured that out :)

**Comment by bgruening** [comment]
@hexylena neat! Let me know when you need another review.

**Comment by hexylena** [comment]
Ok, we think this is ready for review! We've made the following changes, largely based on reviewer comments from the paper submission:

- add built in karyotypes from circos which provide easiest path to get started
- comment out option to use custom cytogenic bands (basically never used, especially now with circos karyotype presets) and was confusing for users.
- Highlights are merged into the 2d data tracks (apparently there are 'highlights' and 'plots; type=highlight' that have very subtle differences.) This is the most universal option.
- Add 2d connector data type 
- Add scaling / zooming feature
- Add limits adjustments (per user issue)
- Added a tool to convert bigwigs to stacked histograms. It only supports the happy path, but this isn't a high-demand feature and it's OK enough for the limited use cases.

And some additional quality of life things:

- grep the error message out, if circos fails due to an internal circos error (xref https://github.com/galaxyproject/galaxy/pull/9690)
- fixed the built in karyotype files since upstream's are broken.
- More sane/pretty defaults.

File: tools/circos/circos.xml [hunk 6]
```diff
@@ -1,14 +1,25 @@
 <?xml version="1.0"?>
 <tool id="circos" name="Circos" version="@WRAPPER_VERSION@">
     <description>visualizes data in a circular layout</description>
+    <xrefs>
+        <xref type="bio.tools">galactic_circos</xref>
```
Comment (bgruening) [inline tools/circos/circos.xml:5]: great idea!

We could add, `<xref type="bioconda">circos</xref>` as well.

File: tools/circos/circos.xml [hunk 7]
```diff
@@ -99,16 +107,16 @@ cp '${data.data_source}' circos/data/highlight-${hi}.txt &&
...
         <expand macro="configfile_links_conf" />
-        <expand macro="configfile_highlight_conf" />
         <expand macro="test_case" />
     </configfiles>
     <inputs>
         <section name="reference_genome" title="Karyotype" expanded="true">
             <conditional name="ref">
                 <param name="ref_source" type="select" label="Reference Genome Source">
-                    <option value="history" selected="True">FASTA File from History</option>
+                    <option value="preset" selected="True"> Circos Presets </option>
```
Comment (bgruening) [inline tools/circos/circos.xml:116]: any reason the have the whitespace the beginning?

File: tools/circos/circos.xml [hunk 8]
```diff
@@ -125,39 +133,171 @@ cp '${data.data_source}' circos/data/highlight-${hi}.txt &&
                      But none of the optinos work well or uniformly, so, no. We're forced to remove this as an option completely.
 
                 -->
+                <when value="preset">
+                    <param name="preset_karyotype" type="select" label="Choose your Circos karyotype" help="In parentheses are the naming conventions for chromosomes used in each karyotype file. These are required for some plot parameters (e.g. filtering chromosomes or specifying rules). The karyotype will also be output along with your plot, for your review.">
+                        <option value="karyotype.arabidopsis.tair10.txt"> Arabidopsis tair10 (chr1,chr2,..)</option>
```
Comment (bgruening) [inline tools/circos/circos.xml:138]: here too, I would remove the spaces at ^ and $

File: tools/circos/circos.xml [hunk 9]
```diff
@@ -125,39 +133,171 @@ cp '${data.data_source}' circos/data/highlight-${hi}.txt &&
...
+            <param name="angle_orientation" type="select" label="Angle orientation" help="Chromosomes are plotted clockwise around the chromosome" >
+                <option value="clockwise">Clockwise</option>
+                <option value="counterclockwise">Counterclockwise</option>
+            </param>
+
+            <section name="ideogram_labels" title="Labels">
+                <param name="show_label" type="boolean" truevalue="yes" falsevalue="no" label="Show Label" checked="true"/>
+                <param type="float" value="0.075" label="Radius" name="radius_offset"/>
+                <param type="integer" value="24" label="Label Font Size" name="size"/>
+                <!--<param type="float" value="10" label="Thickness" name="thickness"/>-->
```
Comment (bgruening) [inline tools/circos/circos.xml:246]: Shout this be still out? If so can you add comments why? So we can fix that in the next version or s?

File: tools/circos/circos.xml [hunk 10]
```diff
@@ -125,39 +133,171 @@ cp '${data.data_source}' circos/data/highlight-${hi}.txt &&
...
+                <param name="parallel" type="boolean" label="Parallel" truevalue="yes" falsevalue="no" checked="true"
+                    help="When set to yes/true, labels will be perpendicular to the tangent of the circle at the location of the label. Otherwise, they will be parallel with the tangent of the circle"/>
+            </section>
+
+            <section name="bands" title="Cytogenic Bands">
+                <param name="show_bands" type="boolean" truevalue="yes" falsevalue="no" label="Show Cytogenic Bands?" checked="true" help="Cytogenic bands can be defined inside the karyotype file"/>
+                <!--<param name="bands" type="data" format="tabular" optional="true" label="Cytogenetic Bands"
```
Comment (bgruening) [inline tools/circos/circos.xml:253]: add comment?

Comment (hexylena) [reply tools/circos/circos.xml:253]: Done!

File: tools/circos/circos.xml (hunk 9 above)
Comment (hexylena) [reply tools/circos/circos.xml]: yep! looks like bad copy paste :(

**Comment by hexylena** [comment]
@bgruening addressed, thanks so much for reviewing it!

**Comment by shiltemann** [comment]
thanks for the review @bgruening!

**Pull request merged**